src/app.js

```diff
@@ -38,52 +41,49 @@ export const echo = (appId, token) => (req, res) => {
     .filter((word) => /^(hello|hey)$/i.test(word)).length)
 
     // Send the echo message
-    send(req.body.spaceId,
+    await send(req.body.spaceId,
       util.format(
         'Hey %s, did you say %s?',
         req.body.userName, req.body.content),
-      token(),
-      (err, res) => {
-        if(!err)
-          log('Sent message to space %s', req.body.spaceId);
-      });
+      token());
+  log('Sent message to space %s', req.body.spaceId);
 };
 
 // Send an app message to the conversation in a space
-const send = (spaceId, text, tok, cb) => {
-  request.post(
-    'https://api.watsonwork.ibm.com/v1/spaces/' + spaceId + '/messages', {
-      headers: {
-        Authorization: 'Bearer ' + tok
-      },
-      json: true,
-      // An App message can specify a color, a title, markdown text and
-      // an 'actor' useful to show where the message is coming from
-      body: {
-        type: 'appMessage',
-        version: 1.0,
-        annotations: [{
-          type: 'generic',
+const send = async (spaceId, text, tok) => {
+  let res;
+  try {
+    res = await post(
+      'https://api.watsonwork.ibm.com/v1/spaces/' + spaceId + '/messages', {
+        headers: {
+          Authorization: 'Bearer ' + tok
+        },
+        json: true,
+        // An App message can specify a color, a title, markdown text and
+        // an 'actor' useful to show where the message is coming from
+        body: {
+          type: 'appMessage',
           version: 1.0,
-
-          color: '#6CB7FB',
-          title: 'Echo message',
-          text: text,
-
-          actor: {
-            name: 'Sample echo app'
-          }
-        }]
-      }
-    }, (err, res) => {
-      if(err || res.statusCode !== 201) {
-        log('Error sending message %o', err || res.statusCode);
-        cb(err || new Error(res.statusCode));
-        return;
-      }
-      log('Send result %d, %o', res.statusCode, res.body);
-      cb(null, res.body);
-    });
+          annotations: [{
+            type: 'generic',
+            version: 1.0,
+
+            color: '#6CB7FB',
+            title: 'Echo message',
+            text: text,
+
+            actor: {
+              name: 'Sample echo app'
+            }
+          }]
+        }
+      });
+    log('Send result %d, %o', res.statusCode, res.body);
+  } 
+  catch(err) {
+    log('Error sending message %o', err);
```

Comment:
Looks like this code silently discards the caught exception. I think it should be re-thrown.